hw/arm/xlnx-versal-virt: modify gem/phy DT nodes

ho28 opened the pull request:
Modify the gem/phy device tree nodes so that VxWorks properly binds the genericPhy driver to the ethernet-phy DT node. Previously, the phy was represented using a fixed-link node that VxWorks does not parse properly.

The compatible string for the gem node has been modified from "cdns,zynqmp-gem" to "cdns,versal-gem". The VxWorks gem driver does not match "cdns,zynqmp-gem".

Note: VxWorks needs additional property "phy-mode" to properly bind the driver against "cdns,versal-gem".

ho28 committed:
hw/arm/xlnx-versal-virt: modify gem/phy DT nodes

Modify the gem/phy device tree nodes so that VxWorks
properly binds the genericPhy driver to the ethernet-phy
DT node. Previously, the phy was represented using a
fixed-link node that VxWorks does not parse properly.

The compatible string for the gem node has been modified
from "cdns,zynqmp-gem" to "cdns,versal-gem". The VxWorks
gem driver does not match "cdns,zynqmp-gem".

Note: VxWorks needs additional property "phy-mode" to
properly bind the driver against "cdns,versal-gem".

Signed-off-by: Nelson Ho <Nelson.Ho@windriver.com>

kay-ge commented on hw/arm/xlnx-versal-virt.c:
```diff
@@ -276,15 +276,22 @@ static void fdt_add_canfd_nodes(VersalVirt *s)
     qemu_fdt_setprop_cell(s->fdt, name, "phandle", phandle);
-    qemu_fdt_setprop(s->fdt, name, "full-duplex", NULL, 0);
-    qemu_fdt_setprop_cell(s->fdt, name, "speed", 1000);
+    qemu_fdt_setprop_cell(s->fdt, name, "fifo-depth", 0x01);
```
Should there be #defines or other constant definitions for 0x01, 0a0a, 0x08, 0x03 (the various constants on this line through line 292)?

ho28 replied:
actually, it looks like these properties are not needed when initializing the genericPhy driver. I had borrowed them from the zynqmp device tree, but on the real hardware that phy binds to the tiTdPhy driver; on the qemu platform the phy is not fully implemented and vxworks falls back to the genericPhy driver.

i'll go ahead and remove these other properties since they have no meaning to the emulated phy.

calvincramer commented:
@ho28 can you show some test results? Manual is fine. I'm not too familiar with what the CI tests test, but it looked like generic arm and generic x86 64.

calvincramer reviewed (approved):
@ho28 showed it working live

ho28 commented:
I made the changes mentioned above, removed the unnecessary properties from the phy node.

I also did some additional reading on the reg property of the phy node. I was under the assumption previously that the reg field (0xc) specified an offset from the regs or ranges presented by the GEM. After consulting the device tree documentation, it turns out the reg field is simply the phy index when specified as a property of a phy. I modified the constants and fields as appropriate.
